build_tools/ci/cpu_comparison/run_test.py

```diff
@@ -127,8 +125,21 @@ def generate_aie_output(
     start = time.monotonic_ns()
     shell_out(compilation_flags, config.output_dir, config.verbose)
     compile_time = time.monotonic_ns() - start
+    if config.verbose:
+        print(f"Time spent in compilation: {compile_time // 1e6} [ms]")
 
     aie_vmfb = config.output_dir / f"{name}_aie.vmfb"
+    if not aie_vmfb.exists():
+        raise RuntimeError(f"Failed to compile {test_file} to {aie_vmfb}")
+
+    return {"vmfb_path": aie_vmfb}
```

Comment:
i prefer [namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple) for these kinds of things (or dataclass). if you don't want to both with defining a type at top level you can do 

```suggestion
    return namedtuple("SOMETHING_THAT_WILL_SHOW_UP_IN_REPR/STR", ["vmfb_path"])(aie_vmfb)
```

then down below you'll just be able to `aie_artefacts.vmfb_path`.

But really I wouldn't go through this until we have actually multiple _artefacts_[^1] and just return `aie_vmfb`

[^1]: see what i did there